nx/lib/nx.ex

```diff
@@ -5176,15 +5176,93 @@ defmodule Nx do
   """
   @doc type: :indexed
   def indexed_add(target, indices, updates) do
+    {target, indices, updates, type} = indexed_op(target, indices, updates)
+
+    impl!(target, indices, updates).indexed_add(%{target | type: type}, target, indices, updates)
+  end
+
+  @doc """
+  Performs an indexed `set` operation on the `target` tensor,
+  setting the `updates` into the corresponding `indices` positions.
+
+  `indices` must be a fully qualified tensor of shape `{n, Nx.rank(target)}`, with `n`
+  being an arbitrary number of indices, while `updates` must have a compatible `{n}` shape.
+
+  ### Examples
+
+      iex> t = Nx.iota({1, 2, 3})
+      #Nx.Tensor<
+        s64[1][2][3]
+        [
+          [
+            [0, 1, 2],
+            [3, 4, 5]
+          ]
+        ]
+      >
+      iex> indices = Nx.tensor([[0, 0, 0], [0, 1, 1], [0, 0, 0], [0, 0, 2], [0, 1, 2]])
```

Comment:
I think we should put repeated indices as a separate example to highlight the behavior